### PR TITLE
JSM: Added TS files for NodeMaterial core classes.

### PR DESCRIPTION
**Pull request by Mugen87**
This PR added the TS declaration files for all core classes of `NodeMaterial`.

If the core files are okay, I'm going to work on the remaining code.

### CODE REVIEW

File: examples/jsm/nodes/core/NodeLib.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  export const nodes: object;
+  export const keywords: object;
+
+  export function add(node: Node): void;
+  export function addKeyword(name: string, callback: () => void, cache?: object): void;
+  export function remove(node: Node): void;
+  export function removeKeyword(name: string): void;
+  export function get(name: string): Node;
+  export function getKeyword(name: string, material: object): any;
```
Comment (Mugen87) [inline examples/jsm/nodes/core/NodeLib.d.ts]: @sunag I'm unsure about `getKeyword()`. This method is called in `FunctionNode.generate()` but with `NodeBuilder` as the second parameter. However, in the `NodeLib` source file the parameter is named `material`. 

In any event, `material` is assigned to the callback function defined in `addKeyword()`. What is the intended parameter of that callback function?

**Comment by Mugen87** [comment]
@sunga I've noticed that many properties like `delta`, `renderer` and `renderTexture` from `NodeFrame` are not explicitly declared in the constructor but when invoking certain methods. For clarity, I thinks it's better to declare all properties in the ctor and initialize them with appropriate values if possible. This makes the code easier to read.

File: examples/jsm/nodes/core/NodeLib.d.ts (hunk 1 above)
Comment (sunag) [reply examples/jsm/nodes/core/NodeLib.d.ts]: Hi @Mugen87 `getKeyword(name: string, material: object)` is outdated this should be `getKeyword(name: string, builder: object)`. 

`builder` argument can be useful to the callback return a node appropriate to a determined material and processes like vertex or fragment shader for example:
https://github.com/mrdoob/three.js/blob/a756b4fd1565248dfebd2e7eadb5cfc60ab986e8/examples/webgl_materials_nodes.html#L2192-L2196
Can be something like:
```javascript
Nodes.NodeLib.addKeyword( 'myCustomUv', function ( builder ) { 
  
 	if ( builder.isShader("vertex") ) return new Nodes.ReflectNode(); 
 	else return /*other node*/;
  
 } ); 
```

**Comment by sunag** [comment]
> @sunag I've noticed that certain properties (like e.g. delta, renderer and renderTexture from NodeFrame) are not explicitly declared in the constructor but only when invoking certain methods. For clarity reasons, I thinks it's better to declare all properties in the ctor and initialize them with appropriate values if possible. This makes the code easier to read.

You think something like?
```javascript
this.delta = 0;

this.renderer = undefined;
this.renderTexture = undefined;
```

I am avoiding use parameters to `renderer` and `renderTexture` since this should be changed per node.

File: examples/jsm/nodes/core/NodeLib.d.ts (hunk 1 above)
Comment (Mugen87) [reply examples/jsm/nodes/core/NodeLib.d.ts]: 👍

**Comment by Mugen87** [comment]
Yeah. We had a similar discussion some time ago in context of loaders (/cc @donmccurdy). The conclusion was that defining all properties in the ctor is better for readability.

Anyway, it's not that urgent. I've just mentioned it so are aware of this situation.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**